### PR TITLE
Implement building cache_images

### PR DESCRIPTION
**Pull request by cevich**
Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

**Comment by cevich** [comment]
@edsantiago and @lsm5 if you don't mind, despite this being a WIP I would appreciate feedback on the `README.md`.  Esp. "The last step" section this PR updates: https://github.com/containers/automation_images/pull/8/files?file-filters%5B%5D=.md  Does this seem like an improvement to the process/workflow?  For example, if you had to update a podman or buildah CI image.

Also, I haven't documented it yet, but the `Makefile` here does allow running ***ALL*** the packer builds locally.  So it's also possible to selectively build one Distro or version, e.g. only Ubutnu.  The only required input is a credentials file (which I can provide as needed).

**Comment by cevich** [comment]
w00t! entire build-chain is working & producing images.

File: cache_images/fedora_packaging.sh [hunk 1]
```diff
@@ -7,35 +7,28 @@
...
-# Install common automation tooling (i.e. ooe.sh)
-curl --silent --show-error --location \
-     --url "https://raw.githubusercontent.com/containers/automation/master/bin/install_automation.sh" | \
-     $SUDO env INSTALL_PREFIX=/usr/share /bin/bash -s - "$INSTALL_AUTOMATION_VERSION"
-# Reload installed environment right now (happens automatically in a new process)
-source /usr/share/automation/environment
+echo "Updating/Installing repos and packages for $OS_REL_VER"
 
 # Set this to 1 to NOT enable updates-testing repository
 DISABLE_UPDATES_TESTING=${DISABLE_UPDATES_TESTING:0}
```
Comment (edsantiago) [inline cache_images/fedora_packaging.sh:20]: Positive variable names, please.

File: cache_images/fedora_packaging.sh [hunk 2]
```diff
@@ -7,35 +7,28 @@
...
 
 # Do not enable updates-testing on the previous Fedora release
 if ((DISABLE_UPDATES_TESTING!=0)); then
```
Comment (edsantiago) [inline cache_images/fedora_packaging.sh:23]: One reason to use positive variable names: it's easier to catch logic bugs like this one.(But the most important reason is that it's a lower cognitive load on maintainers)

File: cache_images/fedora_packaging.sh [hunk 3]
```diff
@@ -159,36 +152,25 @@ DOWNLOAD_PACKAGES=(\
...
-# that is not yet in stable. Since CI:IMG PRs are a two-step process,
-# the key part is that we UN-COMMENT-THIS-OUT during the first step,
-# then re-comment it on the second (once we have the built images).
-# That way this will be dead code in future CI:IMG PRs but will
-# serve as an example for anyone in a similar future situation.
-#   $BIGTO ooe.sh $SUDO dnf --enablerepo=updates-testing -y upgrade crun
+echo "Installing general build/test dependencies"
+bigto ooe.sh $SUDO dnf install -y "${INSTALL_PACKAGES[@]}"
 
 [[ ${#REMOVE_PACKAGES[@]} -eq 0 ]] || \
```
Comment (edsantiago) [inline cache_images/fedora_packaging.sh:158]: Why not make this a more readable `if`?

File: cache_images/fedora_packaging.sh [hunk 4]
```diff
@@ -159,36 +152,25 @@ DOWNLOAD_PACKAGES=(\
...
-    $LILTO ooe.sh $SUDO dnf erase -y "${REMOVE_PACKAGES[@]}"
+    lilto ooe.sh $SUDO dnf erase -y "${REMOVE_PACKAGES[@]}"
 
 if [[ ${#DOWNLOAD_PACKAGES[@]} -gt 0 ]]; then
     echo "Downloading packages for optional installation at runtime, as needed."
     # Required for cri-o
-    ooe.sh $SUDO dnf -y module enable cri-o:$(get_kubernetes_version)
+    ooe.sh ooe.sh $SUDO dnf -y module enable cri-o:$(get_kubernetes_version)
```
Comment (edsantiago) [inline cache_images/fedora_packaging.sh:164]: Do you intentionally want two `ooe.sh`s? If so, can you document why? It's not really clear to me what this does.

File: cache_images/fedora_packaging.sh [hunk 5]
```diff
@@ -159,36 +152,25 @@ DOWNLOAD_PACKAGES=(\
...
-cd $GOSRC
-# Required since initially go was not installed
-source $GOSRC/$SCRIPT_BASE/lib.sh
-echo "Go environment has been setup:"
-go env
-$SUDO make install.tools
-$SUDO $GOSRC/hack/install_catatonit.sh
+echo "Configuring Go environment"
+mkdir -p /var/tmp/go
+export GOPATH=/var/tmp/go
```
Comment (edsantiago) [inline cache_images/fedora_packaging.sh:172]: If you switch these two lines (and, of course, `mkdir -p $GOPATH`) you'll save yourself a maintenance burden if/when `/var/tmp/go` changes to something else.

File: cache_images/fedora_setup.sh [hunk 6]
```diff
@@ -5,30 +5,30 @@
...
 fi
 
-bash $PACKER_BASE/fedora_packaging.sh
-# Load installed environment right now (happens automatically in a new process)
-source /usr/share/automation/environment
+bash $SCRIPT_DIRPATH/fedora_packaging.sh
 
 echo "Enabling cgroup management from containers"
-ooe.sh sudo setsebool container_manage_cgroup true
+ooe.sh $SUDO setsebool container_manage_cgroup true
```
Comment (edsantiago) [inline cache_images/fedora_setup.sh:30]: Should this be `-P`?

File: cache_images/fedora_packaging.sh (hunk 1 above)
Comment (cevich) [reply cache_images/fedora_packaging.sh:20]: this is how it was done previously...but I can change it (it hurts my brain also).

File: cache_images/fedora_packaging.sh (hunk 2 above)
Comment (cevich) [reply cache_images/fedora_packaging.sh:23]: yep...gotcha and gotcha...thx.

File: cache_images/fedora_packaging.sh (hunk 4 above)
Comment (cevich) [reply cache_images/fedora_packaging.sh:164]: woopsies...that was a search-replace goof.

File: cache_images/fedora_packaging.sh (hunk 5 above)
Comment (cevich) [reply cache_images/fedora_packaging.sh:172]: great, got it.  And fixed similar thing in `ubuntu_packaging.sh` also.

File: cache_images/fedora_setup.sh (hunk 6 above)
Comment (cevich) [reply cache_images/fedora_setup.sh:30]: Ooohhhh, GOOD CATCH!  yes!

File: cache_images/fedora_packaging.sh (hunk 3 above)
Comment (cevich) [reply cache_images/fedora_packaging.sh:158]: okay, fair-nuff...got it.

File: Makefile [hunk 7]
```diff
@@ -153,16 +156,17 @@ image_builder/manifest.json: image_builder/gce.json image_builder/setup.sh lib.s
 	$(eval override _GAC_FILEPATH := $(call err_if_empty,GAC_FILEPATH))
 	$(call packer_build,$<)
 
-# TODO: Make this work.  Currently when qemu-kvm starts up, it complains
-# tries and fails to read files from my home directory on the host (outside
-# the container).
+# TODO: Document: Unless we assume $PWD is under the uses home-dir, accessing
+# sometimes necessary files like $HOME/.gitconfig or $HOME/.ssh won't work if
+# the container were to only volume-mount $PWD.
```
Comment (edsantiago) [inline Makefile:161]: (probably for later): I find it hard to parse this comment. If cwd is under HOME, how could you access `$HOME/.anything` by mounting `$PWD`? Would it make sense to comment instead something like `$HOME is needed for .config, .ssh, etc`?

File: .cirrus.yml [hunk 8]
```diff
@@ -8,6 +8,9 @@ env:
     CIRRUS_SHELL: "/bin/bash"
     # No need to go crazy, but grab enough to cover most PRs
     CIRRUS_CLONE_DEPTH: 50
+    IMG_SFX: "${CIRRUS_BUILD_ID}"  # Same for all tasks of _this_ build
+    # Imgae create/import credentials (N/B: must NOT == gcp_credentials)
```
Comment (edsantiago) [inline .cirrus.yml:12]: (for later, not now) typo: imgae. And maybe you could explain what `GAC` stands for.

File: README.md [hunk 9]
```diff
@@ -9,44 +9,30 @@ suitability for use by other *containers* org. repos.
...
-   *Review changes* button within the PR.  Assuming the review is (eventually)
-   marked as 'approved', and the PR is not a *draft*, it will automatically
-   merge when all CI tests pass.
-
-5. This repo. follows the [fork & pull
-   model](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)
-
-6. Strict-merging is enabled to guarantee the tip of the base branch has
-   always been checked by automation.  All pull requests must be kept up
+4. Strict-merging is enabled, therefor all pull requests must be kept up
```
Comment (edsantiago) [inline README.md:24]: (for later) therefore with an 'e'

File: Makefile (hunk 7 above)
Comment (cevich) [reply Makefile:161]: Long-story short: this command _use_ to volume mount `$PWD` only, took me hours to debug related problems.  It's likely  some of that debugging-think crept into my comment.  Yes, something along the lines of your suggestion would be better, but let's do it later so the images don't need to be re-built :grin:

File: .cirrus.yml (hunk 8 above)
Comment (cevich) [reply .cirrus.yml:12]: Google Application Credentials (I swore I spelled it all out somewhere...maybe it got lost in a past revision).  Anyway, it's not my term, it's what google calls the JSON file in their docs also.

File: README.md (hunk 9 above)
Comment (cevich) [reply README.md:24]: *nod*

File: Makefile (hunk 7 above)
Comment (cevich) [reply Makefile:161]: k. got this.

File: .cirrus.yml (hunk 8 above)
Comment (cevich) [reply .cirrus.yml:12]: k, fixed.
